src/Analyzer/Passes/LogicalExpressionOptimizerPass.cpp

```diff
@@ -590,6 +590,9 @@ class JoinOnLogicalExpressionOptimizerVisitor : public InDepthQueryTreeVisitorWi
 
     void enterImpl(QueryTreeNodePtr & node)
     {
+        if (node.use_count() > 1)
+            node = node->clone();
+
```

Comment:
Fuzzed in the PR, but different issue fix from https://github.com/ClickHouse/ClickHouse/pull/74970